x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/document_fields/field_parameters/fielddata_parameter.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiSpacer, EuiFormRow, EuiDualRange, EuiCallOut, EuiLink } from '@elastic/eui';
+
+import {
+  UseField,
+  Field,
+  FormDataProvider,
+  FieldHook,
+  UseMultiFields,
+} from '../../../shared_imports';
+import { getFieldConfig } from '../../../lib';
+import { EditFieldFormRow } from '../fields/edit_field';
+import { documentationService } from '../../../../../services/documentation';
+
+export const FieldDataParameter = () => {
+  const onFrequencyFilterChange = (minField: FieldHook, maxField: FieldHook) => ([
+    min,
+    max,
+  ]: any) => {
+    minField.setValue(min);
+    maxField.setValue(max);
+  };
+
+  return (
+    <EditFieldFormRow
+      title={i18n.translate('xpack.idxMgmt.mappingsEditor.fielddata.fielddataFormRowTitle', {
+        defaultMessage: 'Fielddata',
+      })}
+      description={i18n.translate(
+        'xpack.idxMgmt.mappingsEditor.fielddata.fielddataFormRowDescription',
+        {
+          defaultMessage:
+            'Whether to use in-memory fielddata for sorting, aggregations, or scripting.',
+        }
+      )}
+      formFieldPath="fielddata"
+    >
+      <FormDataProvider pathsToWatch="fielddata">
```

Comment:
nit: It would be better to just wrap the JSX that is interested by the change. So here:

```js
<FormDataProvider pathsToWatch="fielddata">
{({ fielddata }) => (
  fielddata === true ? (
    <>
      <EuiSpacer size="s" />
      <EuiCallOut color="warning">
        <p>
          <FormattedMessage
            id="xpack.idxMgmt.mappingsEditor.fielddata.fielddataEnabledWarningMessage"
            defaultMessage="Fielddata can consume significant memory, especially when loading high-cardinality text fields. {docsLink}"
            values={{
              docsLink: (
                <EuiLink
                  href={documentationService.getEnablingFielddataLink()}
                  target="_blank"
                >
                  {i18n.translate(
                    'xpack.idxMgmt.mappingsEditor.fielddata.fielddataEnabledDocumentationLink',
                    {
                      defaultMessage: 'Learn more.',
                    }
                  )}
                </EuiLink>
              ),
            }}
          />
        </p>
      </EuiCallOut>
    </>
  ) : null)}
</FormDataProvider>
```